ruby/src/IceRuby/Communicator.cpp

```diff
@@ -188,7 +188,12 @@ IceRuby_initialize(int argc, VALUE* argv, VALUE /*self*/)
         seq.insert(seq.begin(), getString(progName));
 
         data.compactIdResolver = [](int id) { return IceRuby::resolveCompactId(id); };
-        data.valueFactoryManager = ValueFactoryManager::create();
+
+        ValueFactoryManagerPtr valueFactoryManager =  ValueFactoryManager::create();
```

Comment:
This code was problematic because `ValueFactoryManager::create()` creates a Ruby object in the HEAP. The object is not referenced by any other Ruby objects until the communicator is created.

Later in this methods we make some ruby calls before creating the communicator, and any of them can trigger a GC and free this object.